src/rgw/rgw_rest_s3.cc

```diff
@@ -1705,11 +1705,13 @@ int RGWPostObj_ObjStore_S3::get_policy()
       err_msg = "Missing signature";
       return -EINVAL;
     }
-
     RGWUserInfo user_info;
 
     op_ret = rgw_get_user_info_by_access_key(store, s3_access_key, user_info);
     if (op_ret < 0) {
+      S3AuthFactory aplfact(store, s->account_name);
+      RGWLDAPTokenExtractor token_extr(s);
```

Comment:
RGWLDAPAuthEngine::applicable has all the logic to determine whether ldap is correctly configured to be used or not including checking the configurable. Moreover the extractor will be common to ldap, keystone and local auth, so may end up eventually getting renamed and one object of the extractor will be defined for all three auth engines.(This depends on the review of local auth rework code.)